analysis/crossref/crossref.go

```diff
@@ -31,27 +34,31 @@ func BuildXRefCollection(res *database.Resources) {
 		external[selector.GetName()] = externalHosts
 	}
 
-	//build internal and external at the same time
-	//we could build the two collections at the same time
-	//but, we have a thread for each analysis module reading,
-	//this thread, and a number of write threads already spun.
-	//TODO: config collection names
-	multiplexXRef(res, res.System.CrossrefConfig.InternalTable, internal)
-	multiplexXRef(res, res.System.CrossrefConfig.ExternalTable, external)
+	xRefWG := new(sync.WaitGroup)
+	xRefWG.Add(2)
+	//kick off writes
+	go multiplexXRef(res, res.System.CrossrefConfig.InternalTable, internal, xRefWG)
```

Comment:
In order to fix the deadlock, we have to build the internal and external crossref collections at the same time.